immuni_exposure_ingestion/tasks/process_uploads.py

```diff
@@ -60,70 +66,154 @@ async def _process_uploads() -> None:
     # Acquire a lock on redis before processing anything, avoiding concurrent tasks.
     async with lock_concurrency("process_uploads"):
         _LOGGER.info("Obtained lock.")
+        _LOGGER.info("Start processing italian Teks.")
+        _batch_it()
+        _LOGGER.info("End processing italian Teks.")
+        _LOGGER.info("Start processing EU Teks marked as italian.")
+        _batch_eu()
+        _LOGGER.info("End processing EU Teks marked as italian.")
+        _LOGGER.info("Releasing lock.")
+
+    _LOGGER.info("Upload processing completed successfully.")
 
-        infos = BatchFile.get_latest_info()
-        now = datetime.utcnow()
 
-        if infos:
-            last_period, last_index = infos
-        else:
-            last_period = datetime.fromtimestamp(
-                croniter(config.BATCH_PERIODICITY_CRONTAB).get_prev()
+def _batch_it():
+    """
+    Get the unprocessed uploads from the upload collection, performs some validations and create
+    multiple batches stored in the batch_file collection.
```

Comment:
```suggestion
    a batch. When the maximum number of keys for a single batch is reached, create
    the batch, and leave the remaining keys to be processed on another run of this
    method.
    
    NOTE: The current maximum, and the number of nowadays keys do not require the
      generation of multiple batches in a single run of this method.
      This assumption may fall in the future, and we should monitor no queue of
      unprocessed keys will ever get stuck.
```

I'd change the EU method too.